### ISSUE TITLE
backslash in string incorrectly escaping closing quote

### ISSUE THREAD

**Issue by aharpervc** [atom/language-sql#13]
Example:
![wrong](http://i.imgur.com/VIi17Gv.png)

It should look like this (minus the space I added to hack the highlighting):
![right](http://i.imgur.com/Ije2Bxd.png)

**Comment by jlewis91** [comment]
This can likely be closed.

**Comment by aharpervc** [comment]
@jlewis91 it's still happening in 0.17.0. It has not yet been fixed.

**Comment by jlewis91** [comment]
For when this is implemented, here's the spec (courtesy of @C-Bouthoorn):

```
  it "quotes strings", ->
    {tokens} = grammar.tokenizeLine('"Test"')
    expect(tokens[0]).toEqual value: '"', scopes: ['source.sql', 'string.quoted.double.sql', 'punctuation.definition.string.begin.sql']
    expect(tokens[1]).toEqual value: 'Test', scopes: ['source.sql', 'string.quoted.double.sql']
    expect(tokens[2]).toEqual value: '"', scopes: ['source.sql', 'string.quoted.double.sql', 'punctuation.definition.string.end.sql']
```

**Comment by C-Bouthoorn** [comment]
Should it? You escape the quote, so it's still a string, right?

**Comment by aharpervc** [comment]
`\` is not a string escape character in SQL

**Comment by jlewis91** [comment]
In Mysql it is. Looks like this is a parameter that can be set, but as a default, backslash can be used as an escape. See: https://dev.mysql.com/doc/refman/5.0/en/string-literals.html, look for `NO_BACKSLASH_ESCAPES`. Recommend leaving the behavior as it is and closing this.

**Comment by 50Wliu** [comment]
Ok, since we're on this topic now: should language-sql support MySQL or just strictly SQL?  There's another PR open right now (#16) that aims to remove MySQL comments.

**Comment by C-Bouthoorn** [comment]
@aharpervc I'm sorry, I was confused with MySQL

@50Wliu I think we should have a language-sql and language-mysql. MySQL has more differences from default SQL, of which this is one. See [this page](https://dev.mysql.com/doc/mysql-reslimits-excerpt/5.5/en/differences-from-ansi.html) for differences. The struggle is how we're going to see when to use which one.

**Comment by jlewis91** [comment]
Gotcha. Would it be possible to add a preferences toggle to select the sql flavor? It seems like it would be inefficient (also lousy from a user perspective) to fork out MySQL, given the large degree of shared components...

**Comment by 50Wliu** [comment]
> Would it be possible to add a preferences toggle to select the sql flavor?

I don't believe so, though it is possible to have multiple grammars in the same package (see [language-c](https://github.com/atom/language-c/tree/master/grammars) for a good example of this).

**Comment by gontadu** [comment]
Are the various grammars then distinguished based on the file's extension?

**Comment by 50Wliu** [comment]
@gontadu Correct.
